packages/host/app/services/store.ts

```diff
@@ -869,13 +918,14 @@ export default class StoreService extends Service implements StoreInterface {
   private async persistAndUpdate(
     instance: CardDef,
     defaultRealmHref?: string,
-  ): Promise<void> {
-    await this.withTestWaiters(async () => {
+  ): Promise<CardDef | CardErrorJSONAPI> {
+    return await this.withTestWaiters(async () => {
       let cardChanged = false;
       function onCardChange() {
         cardChanged = true;
       }
       let api: typeof CardAPI | undefined;
+      let isNew = !instance.id;
```

Comment:
So in the original case described in the ticket, where the card is created but could not be rendered, returning `isNew` to be `true` means `cardError.meta.isCreationError` will be `true` and playground will display `Failed to create card` message, which seems wrong. Only because the file does actually appear in the system. 

If this will be unnecessarily tricky to sort out, we could also just get rid of that message and just say there's an error.

Reply:
Yeah, I was just trying to be true to the current functionality as there was another test that relied on this. In the scenario you mentioned I think I can have this flag be false.